src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,36 +23,77 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person identified by the index number used in the displayed person list "
+            + "or matriculation number.\n"
+            + "Parameters: INDEX (must be a positive integer) or STUDENT_ID\n"
+            + "Example: " + COMMAND_WORD + " 1"
+            + " or " + COMMAND_WORD + " " + PREFIX_ID + "A0123456Z \n";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
     private final Index targetIndex;
+    private final StudentId targetId;
 
+    /**
+     * Creates a DeleteCommand to delete the specified {@code Person} using the specfied index.
+     */
     public DeleteCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
+        this.targetId = null;
+    }
+
+    /**
+     * Creates a DeleteCommand to delete the specified {@code Person} using the specfied student id.
+     */
+    public DeleteCommand(StudentId targetId) {
+        this.targetId = targetId;
+        this.targetIndex = null;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (targetIndex != null) { // index was used for the command
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deletePerson(personToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+        } else { // student id was used for the command
+            assert targetId != null;
+            String[] idKeywords = targetId.toString().split("\\s+");
+            StudentIdContainsKeywordsPredicate pred = new StudentIdContainsKeywordsPredicate(Arrays.asList(idKeywords));
```

Comment:
```suggestion
            StudentIdContainsKeywordsPredicate pred = new StudentIdContainsKeywordsPredicate(Collections.singletonList(targetId.toString()));
```
I think we don't have to split by whitespaces since our studentId does not have any whitespaces (unlike Name)

For this comment, I think it would be best to not use the commit suggestion function since Collections have to be imported at the start.